sql/modules/Reconciliation.sql

```diff
@@ -93,6 +115,30 @@ $$Sets which lines of the report are cleared.$$;
 CREATE OR REPLACE FUNCTION reconciliation__delete_my_report(in_report_id int)
 RETURNS BOOL AS
 $$
+DECLARE
+    failed BOOL;
+BEGIN
+    SELECT approved INTO failed FROM cr_report
+    WHERE id = in_report_id AND approved;
+    IF FOUND THEN
+        RAISE EXCEPTION 'Cannot delete approved recon report';
+        RETURN FALSE;
+    END IF;
+
+    -- Make sure that transactions present on this report do not have a
+    -- cleared_on date or they won't ever be accessible anymore.
```

Comment:
If "posting" the report sets the cleared date, and posted reports can't be deleted, how come we may fail this test?

Reply:
Submit 2 reports, Refuse the first, Fix it by adding a transaction that was in the 2nd and you have an approved transaction in 2 reports and you can't remove it from the 2nd.
Batch submits in migration make ithat situation easy to arise.